### PR TITLE
Fix test error in InternalCardinalityTests#testEqualsAndHashcode

### LINKED ISSUE #62540: InternalCardinalityTests#testEqualsAndHashcode failure
The failure can easily be reproduce with:

```
 ./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.metrics.InternalCardinalityTests.testEqualsAndHashcode" -Dtests.seed=7FFFBA60D5055C4B -Dtests.security.manager=true -Dtests.locale=ar-EG -Dtests.timezone=Europe/Saratov -Druntime.java=11
```

And the error looks like:

```
  2> java.lang.AssertionError: InternalCardinality mutation should not be equal to original
    Expected: not <{"fmTtJ":{"meta":{"RPdFM":"Bglpg","VLYUl":"qWMAU"},"value":50}}>
         but: was <{"fmTtJ":{"meta":{"RPdFM":"Bglpg","VLYUl":"qWMAU"},"value":50}}>
        at __randomizedtesting.SeedInfo.seed([7FFFBA60D5055C4B:EF0C2AD1AE21564]:0)

```

I had a look and the problem is as follows. Before #62480 for testing equality we were ignoring the provided bucket and were generating a string that was generated by all the elements in the runLens array. From now on we are only considering the runLens in the bucket (which I think is the right thing).

The test is mutating a HyperLogLog++ by adding new elements, unfortunately the values did not change so the resulting sketch is the same as the previous one. Before it did not fail because the original  HyperLogLog++ contained more buckets.

### PR DESCRIPTION
**Pull request by iverase**
Make sure the the new HLL++ is different to the original one by not merging the original and just add a few values to the new one.

closes #62540

### CODE REVIEW

**Comment by iverase** [comment]
@elasticmachine elasticsearch-ci/packaging-sample-windows

**Comment by iverase** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-windows

**Comment by iverase** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-windows

File: server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalCardinalityTests.java [hunk 1]
```diff
@@ -99,9 +99,7 @@ protected InternalCardinality mutateInstance(InternalCardinality instance) {
         case 1:
             HyperLogLogPlusPlus newState = new HyperLogLogPlusPlus(state.precision(),
                     new MockBigArrays(new MockPageCacheRecycler(Settings.EMPTY), new NoneCircuitBreakerService()), 0);
-            newState.merge(0, state, 0);
-            int extraValues = between(10, 100);
-            for (int i = 0; i < extraValues; i++) {
+            for (int i = 0; i < 10; i++) {
                 newState.collect(0, BitMixer.mix64(randomIntBetween(500, 10000)));
```
Comment (polyfractal) [inline server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalCardinalityTests.java:103]: Haven't looked closely at the test, but is there a way to ensure we add new values that weren't previously added?  Maybe by picking from a different range of random values or something?  Or keeping a set? 

Maybe it's a non-issue and random enough to not matter... might be a problem for a different day :)

**Review by polyfractal — APPROVED** [review]
Left a comment about a maybe-future tweak, but LGTM as it stands right now :)

File: server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalCardinalityTests.java (hunk 1 above)
Comment (iverase) [reply server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalCardinalityTests.java:103]: It does use a different range of values, Still for precision 4 in the previous approach you could get the same sketch at the end :( I think this way is very very unlikely.

Comment (polyfractal) [reply server/src/test/java/org/elasticsearch/search/aggregations/metrics/InternalCardinalityTests.java:103]: Ah gotcha.  Sounds good!

**Pull request merged**